packages/meditrak-server/src/routes/utilities/SurveyResponseVariablesExtractor.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Tupaia MediTrak
+ * Copyright (c) 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { ValidationError } from '@tupaia/utils';
+
+/**
+ * A function that extracts `entityId` and `answers` from a row
+ *
+ * @callback ResponseExtractor
+ * @async
+ * @param {string[]} row
+ * @returns {{ entityId, answers }}
+ */
+
+export class SurveyResponseVariablesExtractor {
+  /**
+   * @param {ModelRegistry} models
+   */
+  constructor(models) {
+    this.models = models;
+  }
+
+  /**
+   * @public
+   * @param {Object<string, Array>} rowsBySurvey
+   * @param {string} userId
+   * @returns {Promise<{country: *, countryId: *}>}
+   */
+  async getVariablesByCountryCode(countryCode) {
+    const country = await this.models.country.findOne({ code: countryCode });
+    return { country, countryId: country.id };
+  }
+
+  async getVariablesByEntityCode(entityCode) {
+    const entity = await this.models.entity.findOne({ code: entityCode });
+    const country = await this.models.country.findOne({ code: entity.country_code });
+    return { country, entities: [entity] };
```

Comment:
I reckon these should all have a consistent return type if possible (`{country, countryId, entities}` or just `{ country, entities }`, plus `surveyResponse` and `surveyId` optionally). You could use [typedef](https://jsdoc.app/tags-typedef.html) to ensure this consistency is documented, but it's not essential.

Reply:
Good suggestion! Didn't know any jsdoc before. It is a great tool to highlight the requirement. It's worth to Spend some time to implement it.